console/src/tasks.rs

```diff
@@ -114,8 +136,23 @@ impl State {
         if let Some(now) = update.now {
             self.last_updated_at = Some(now.into());
         }
+
+        if let Some(new_metada) = update.new_metadata {
+            let metas: HashMap<_, Metadata> = new_metada
+                .metadata
+                .into_iter()
+                .map(|meta| {
+                    let id = meta.id.expect("no id").id;
+                    let metadata = meta.metadata.expect("no metadata");
+                    (id, metadata.into())
+                })
+                .collect();
+            self.metas.extend(metas);
```

Comment:
nit: [the `extend` method takes a `T: IntoIterator`](https://doc.rust-lang.org/std/iter/trait.Extend.html#tymethod.extend), and [all types implementing `Iterator` also implement `IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html#impl-IntoIterator-25). so, we don't need to call `collect` here and could skip allocating an intermediate hashmap --- making the code slightly simpler and removing an unnecessary allocation.

```suggestion
            let metas = new_metada
                .metadata
                .into_iter()
                .map(|meta| {
                    let id = meta.id.expect("no id").id;
                    let metadata = meta.metadata.expect("no metadata");
                    (id, metadata.into())
                });
            self.metas.extend(metas);
```